policy/policy.go

```diff
@@ -77,9 +77,9 @@ func (c *clist[T]) Clear() {
 	clear(c.keys)
 }
 
-func (c *clist[T]) AllReverse() iter.Seq[*list.Element[T]] {
-	return func(yield func(*list.Element[T]) bool) {
-		for e := c.l.Back(); e != nil; e = e.Prev() {
+func (c *clist[T]) AllReverse() iter.Seq[*internal.Element[T]] {
```

Comment:
Maybe not ideal to have an exported iterator that returns an internal thing.

Reply:
I guess also applies to other methods here too. Re your other comment, maybe all this should move into internal.

Reply:
Well internal.Element is export to here. And clist isn't exported, so usage remains here.